brain/compartmentReportReader.cpp

```diff
@@ -0,0 +1,203 @@
+/* Copyright (c) 2017, EPFL/Blue Brain Project
+ *                     Juan Hernando <juan.hernando@epfl.ch>
+ *                     Mohamed-Ghaith Kaabi <mohamed.kaabi@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "compartmentReportReader.h"
+#include "detail/compartmentReportReader.h"
+
+#include <iostream>
+
+namespace brain
+{
+CompartmentReportReader::CompartmentReportReader(const brion::URI& uri)
+    : _impl{new detail::CompartmentReportReader(uri)}
+{
+}
+
+CompartmentReportReader::~CompartmentReportReader()
+{
+}
+
+const CompartmentReportMetaData& CompartmentReportReader::getMetaData() const
+{
+    return _impl->metaData;
+}
+
+CompartmentReportView CompartmentReportReader::createView(
+    const brion::GIDSet& cells)
+{
+    return CompartmentReportView(_impl, cells);
+}
+
+CompartmentReportFrame::CompartmentReportFrame()
+    : _impl{new detail::CompartmentReportFrame}
+{
+}
+
+CompartmentReportFrame::~CompartmentReportFrame()
+{
+}
+
+CompartmentReportFrame::CompartmentReportFrame(
+    CompartmentReportFrame&& other) noexcept : _impl(std::move(other._impl))
```

Comment:
It is defined in both files. As it is part of the function signature, it looks weird to not put it in the cpp file too.